lib/ci/lib/rails_upgrade_analyzer.rb

```diff
@@ -0,0 +1,11 @@
+require "octokit"
+require "travis"
+
+$LOAD_PATH.unshift(File.dirname(__FILE__))
+
+module RailsUpgradeAnalyzer
+  autoload :AuthenticationError, "rails_upgrade_analyzer/github_proxy"
+  autoload :GithubProxy, "rails_upgrade_analyzer/github_proxy"
+  autoload :JobResult, "rails_upgrade_analyzer/job_result"
+  autoload :ResultReporter, "rails_upgrade_analyzer/result_reporter"
```

Comment:
If I remember right, Matz isn't a fan of `autoload` (he'd like to remove it), so we should maybe avoid using it in new code...